modules/search/client/components/SearchUsers.component.js

```diff
@@ -91,8 +92,10 @@ class SearchUsers extends React.Component {
     this.setState({ resolved: false });
     console.log('Searching for: ' + this.state.searchQuery);
 
+    // eslint-disable-next-line react/no-access-state-in-setstate
     const userResults = await searchUsers(this.state.searchQuery);
```

Comment:
I'm getting this eslint error in that line:
`95:43  error  Use callback in setState when referencing the previous state  react/no-access-state-in-setstate`
I have no idea on how to fix it. Any idea @simison @mrkvon ?
This is the eslint docs about it: https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-access-state-in-setstate.md

Reply:
This is quite peculiar.
This rule takes care specifically of:
```jsx
// incorrect, can lead to state inconsistency
this.setState({ isFoo: !this.state.isFoo }); // flipping a random boolean

// correct
this.setState((previousState) => ({ isFoo: !previousState.isFoo }));
```
@Akronix Your link is describing the reasons for this.

Perhaps try
```
this.setState(() => ({ ... }))
```
but it shouldn't be necessary.

In a while, if not resolved, I can try locally.

Reply:
Still there? For me `npm run lint` passes.

Reply:
Two lines below, use `this.setState(() => ({ resolved: true, users: userResults.data }))` instead. `const userResults = ...` references `this.state` and that's what this rule is worried about.

I haven't wrapped my head around this particular example, but at least the lint error disappears! :slightly_smiling_face:

Reply:
I think this section should be structured like this for it to work as expected and make the linter happy (untested):

```js
this.setState({ resolved: false }, async () => {
  searchUsers(this.state.searchQuery)
    .then((data) => {
      // Do something in success
    })
    .catch((error) => {
      // Do something in errors
    })
    .finally(() => {
      this.setState({
        resolved: true,
      });
    });
});
```

Reply:
That said, perhaps kinda `resolved` (I can regognice Angularism) could be renamed to `isQuerying`, `isSearching` or similar?